src/Microsoft.TestPlatform.Client/Discovery/DiscoveryRequest.cs

```diff
@@ -355,6 +377,47 @@ public void HandleLogMessage(TestMessageLevel level, string message)
         /// <param name="rawMessage">Raw message.</param>
         public void HandleRawMessage(string rawMessage)
         {
+            if (this.requestData.IsTelemetryOptedIn)
+            {
+                var message = this.dataSerializer.DeserializeMessage(rawMessage);
+
+                if (string.Equals(message.MessageType, MessageType.DiscoveryComplete))
+                {
+                    var discoveryCompletePayload =
+                        this.dataSerializer.DeserializePayload<DiscoveryCompletePayload>(message);
+
+                    if (discoveryCompletePayload != null)
+                    {
+                        if (discoveryCompletePayload.Metrics == null)
+                        {
+                            discoveryCompletePayload.Metrics = new Dictionary<string, object>();
+                        }
+
+                        foreach (var kvp in this.requestData.MetricsCollection.Metrics)
+                        {
+                            discoveryCompletePayload.Metrics.Add(kvp.Key, kvp.Value);
+                        }
+
+                        var discoveryFinalTimeTakenForDesignMode = DateTime.UtcNow - this.discoveryStartTime;
+
+                        // Collecting Total Time Taken
+                        discoveryCompletePayload.Metrics[TelemetryDataConstants.TimeTakenInSecForDiscovery] = discoveryFinalTimeTakenForDesignMode.TotalSeconds;
+                    }
+
+                    int version = 2;
```

Comment:
If there's no Version from upward, you have to send without version right ?

Reply:
Fixed.